### DISCUSSION TITLE
Nonces - Headless Woocommerce Store Question

### DISCUSSION THREAD

**Discussion by TheAggressive** [woocommerce/woocommerce#45005] [WooCommerce, Blocks, and Site Editing]
I'm currently working on a NextJS Headless Woocommerce store using the StoreApi. Im trying to store the Cart-Token and nonce in a httpOnly cookie in middleware on first visit to use for further actions (add item to cart, etc). 

How long do nonce keys live for? I saw on wordpress docs it says those last anywhere from 12 hours - 24 hours but wanted to make sure so I could set an expire time and update when needed by calling the cart endpoint and getting the header info (/wp-json/wc/store/v1/cart). This is currently the only way I know of on how to get that info without being on the WordPress PHP end of things.

If any better solutions, please let me know. Thank you very much for you help!

**Comment by ralucaStan** [comment]
Hey @TheAggressive 
you can read more about WordPress nonces expire in this [Understand and use WordPress nonces properly](https://developer.wordpress.org/news/2023/08/understand-and-use-wordpress-nonces-properly/#what-if-i-tell-you-that-wordpress-nonce-is-not-a-nonce).

**Comment by senadir** [comment]
As Raluca shared, nonces do change depending on the session state, you get a different if a user is logged in or not, if the cart is empty or not.

Nonces are sent with all Store API requests, so it's best to have a middleware that also keeps it update on each request, a safe expiry date is 12 hours after which you can refresh the nonce.
